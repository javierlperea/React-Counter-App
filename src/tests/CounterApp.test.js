import React from 'react';
const { shallow } = require("enzyme")
const { default: CounterApp } = require("../CounterApp")


describe('Pruebas en <CounterApp />', () => {
    
    let wrapper = shallow( <CounterApp /> );
    //esta funcion se repite antes de cada test
    beforeEach( () => {
        wrapper = shallow( <CounterApp /> );
    });

    test('Debe mostrar <CounterApp /> correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar el valor por defecto 100', () => {

        const wrapper = shallow( <CounterApp value={100} />);
        const counterText = wrapper.find('h2').text().trim();  // <--- al texto del h2 le elimina los espacios
        expect( counterText ).toBe('Counter = 100');
    });
    
    test('Debe incrementar con el boton add', () => {
        //busca los btn segun su posicion 0,1,2
        //con simulate puedo simular el evento click sobre el btn
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect( counterText ).toBe('Counter = 11');
    });

    test('Debe decrementar con el boton remove', () => {

        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect( counterText ).toBe('Counter = 9');
    });
    
    test('Debe colocar el valor por defecto con reset', () => {
        
        const wrapper = shallow( <CounterApp value={105} />);

        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('Counter = 105')
    })
    

    
})
