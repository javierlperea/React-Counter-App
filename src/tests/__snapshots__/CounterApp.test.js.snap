// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    Counter = 
    10
  </h2>
  <button
    className="btn blue"
    onClick={[Function]}
  >
    Remove
  </button>
  <button
    className="btn red"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    className="btn green"
    onClick={[Function]}
  >
    Add
  </button>
</Fragment>
`;
